File: src/com/qdacity/umleditor/UmlCodePosition.java
package com.qdacity.umleditor;

import javax.jdo.annotations.IdGeneratorStrategy;
import javax.jdo.annotations.IdentityType;
import javax.jdo.annotations.PersistenceCapable;
import javax.jdo.annotations.Persistent;
import javax.jdo.annotations.PrimaryKey;

/**
 * Represents the position of a code in the uml editor.
 *
 */
@PersistenceCapable(
	identityType = IdentityType.APPLICATION)
public class UmlCodePosition {

	@PrimaryKey
	@Persistent(valueStrategy = IdGeneratorStrategy.IDENTITY)
	private Long id;

	@Persistent
	private Long codeId;

	@Persistent
	private Long codeSystemId;
	
	@Persistent
	private double x;

	@Persistent
	private double y;
	
	public UmlCodePosition() {
		
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getCodeId() {
		return codeId;
	}

	public void setCodeId(Long codeId) {
		this.codeId = codeId;
	}

	public Long getCodeSystemId() {
		return codeSystemId;
	}

	public void setCodeSystemId(Long codeSystemId) {
		this.codeSystemId = codeSystemId;
	}

	public double getX() {
		return x;
	}

	public void setX(double x) {
		this.x = x;
	}

	public double getY() {
		return y;
	}

	public void setY(double y) {
		this.y = y;
	}
}

File: war/assets/js/pages/uml-editor/UmlCodePosition.js
export default class UmlCodePosition {

	constructor() {
		this.codeId = null;

		this.codesystemId = null;

		this.x = 0;

		this.y = 0;
	}

	getCodeId() {
		return this.codeId;
	}

	setCodeId(codeId) {
		this.codeId = codeId;
	}

	getCodesystemId() {
		return this.codesystemId;
	}

	setCodesystemId(codesystemId) {
		this.codesystemId = codesystemId;
	}

	getX() {
		return this.x;
	}

	setX(x) {
		this.x = x;
	}


	getY() {
		return this.y;
	}

	setY(y) {
		this.y = y;
	}
}